src/layouts/docs/utils/add-full-paths-to-nav-data.ts

```diff
@@ -0,0 +1,48 @@
+import { NavNode } from '@hashicorp/react-docs-sidenav/types'
+import { MenuItem } from 'components/sidebar'
+
+// TODO: export NavBranch and NavLeaf
+// TODO: types from react-docs-sidenav.
+// TODO: isNavBranch & isNavLeaf might
+// TODO: also make sense to include in the
+// TODO: react-docs-sidenav component,
+// TODO: maybe in the types.ts file as well.
+interface NavBranch {
+  title: string
+  routes: NavNode[]
+}
+
+interface NavLeaf {
+  title: string
+  path: string
+}
+
+function isNavBranch(value: NavNode): value is NavBranch {
+  return value.hasOwnProperty('routes')
+}
+
+function isNavLeaf(value: NavNode): value is NavLeaf {
+  return value.hasOwnProperty('path')
+}
```

Comment:
I currently don't agree that we should be importing from `react-docs-sidenav` because my current understanding is that the code we're writing is meant to replace that code, but let's add that as another topic to a discussion we have following this PR. Not a blocker for merging. 👍🏻

Reply:
Sounds good - yeah just reaching for what's available, very open to updating with new code 👍